### PR TITLE
Revert Dispatcher changes

### PR DESCRIPTION
**Pull request by jbagga**
Addresses aspnet/Home#2741

### CODE REVIEW

File: src/Microsoft.AspNetCore.Routing/RouteCollection.cs
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Routing/RouteCollection.cs]: Could you revert this file?

**Review by pranavkm — APPROVED** [review]
Seems ok to me. 

1. Could you re-open https://github.com/aspnet/Routing/issues/441 and triage it for preview1? 
2. Build Mvc using these changes and verify things continue working?

**Comment by pranavkm** [comment]
Do get signoff from @rynowak

**Comment by jbagga** [comment]
Verified MVC builds locally, reopened #441 and reverted RouteCollection

**Comment by jbagga** [comment]
@rynowak Updated with changes Pranav suggested. Waiting for your sign off

**Comment by jbagga** [comment]
@rynowak Updated. You should expect to see that the Routing benchmarks project has moved from test to benchmarks directory.

**Review by rynowak — APPROVED** [review]

**Pull request merged**
